ChatSecure/Classes/View Controllers/OTRMessagesViewController.m

```diff
@@ -1945,4 +1929,37 @@ - (IBAction)didPressMigratedSwitch {
     }
 }
 
+// If we find any incoming migration link messages, show the "your friend has migrated" header
+// to allow the user to start chatting with the new account instead.
+- (void) checkRangeForMigrationMessage:(NSRange)range {
```

Comment:
It's ok to call messageAtIndexPath within the methods that are dequeueing cells and whatnot from the main thread, but the issue was that the global dispatch queue will be calling it from all sorts of background threads, so at times messageAtIndexPath will block the other threads (including the main thread). 

The in-app link handling and delegate code is fine. 👍